source/StarField.cpp

```diff
@@ -130,109 +130,119 @@ void StarField::Draw(const Point &pos, const Point &vel, double zoom, const Syst
 
-	double baseZoom = zoom;
-
 	// Check preferences for the parallax quality.
 	const auto parallaxSetting = Preferences::GetBackgroundParallax();
 	int layers = (parallaxSetting == Preferences::BackgroundParallax::FANCY) ? 3 : 1;
 	bool isParallax = (parallaxSetting == Preferences::BackgroundParallax::FANCY ||
 						parallaxSetting == Preferences::BackgroundParallax::FAST);
 
+	double baseZoom = isParallax ? zoom * 0.8 : zoom;
+
 	// Draw the starfield unless it is disabled in the preferences.
 	if(Preferences::Has("Draw starfield") && density > 0.)
 	{
-		glUseProgram(shader.Object());
-		glBindVertexArray(vao);
-
 		for(int pass = 1; pass <= layers; pass++)
 		{
-			// Modify zoom for the first parallax layer.
-			if(isParallax)
-				zoom = baseZoom * STAR_ZOOM * pow(pass, 0.2);
-
-			float length = vel.Length();
-			Point unit = length ? vel.Unit() : Point(1., 0.);
-			// Don't zoom the stars at the same rate as the field; otherwise, at the
-			// farthest out zoom they are too small to draw well.
-			unit /= pow(zoom, .75);
-
-			float baseZoom = static_cast<float>(2. * zoom);
-			GLfloat scale[2] = {baseZoom / Screen::Width(), -baseZoom / Screen::Height()};
-			glUniform2fv(scaleI, 1, scale);
-
-			GLfloat rotate[4] = {
-				static_cast<float>(unit.Y()), static_cast<float>(-unit.X()),
-				static_cast<float>(unit.X()), static_cast<float>(unit.Y())};
-			glUniformMatrix2fv(rotateI, pass, false, rotate);
-
-			glUniform1f(elongationI, length * zoom);
-			glUniform1f(brightnessI, min(1., pow(zoom, .5)));
-
-			// Stars this far beyond the border may still overlap the screen.
-			double borderX = fabs(vel.X()) + 1.;
-			double borderY = fabs(vel.Y()) + 1.;
-			// Find the absolute bounds of the star field we must draw.
-			int minX = pos.X() + (Screen::Left() - borderX) / zoom;
-			int minY = pos.Y() + (Screen::Top() - borderY) / zoom;
-			int maxX = pos.X() + (Screen::Right() + borderX) / zoom;
-			int maxY = pos.Y() + (Screen::Bottom() + borderY) / zoom;
-			// Round down to the start of the nearest tile.
-			minX &= ~(TILE_SIZE - 1l);
-			minY &= ~(TILE_SIZE - 1l);
-
-			for(int gy = minY; gy < maxY; gy += TILE_SIZE)
+			Point backgroundPos = (isParallax ? (pos / 2) : pos) + Point(pass * Screen::Width(), pass * Screen::Height());
+			Point backgroundVel = vel / 2;
+
+			// Draw the star layer
 			{
-				float shove = pow(-5., pass);
-				for(int gx = minX; gx < maxX; gx += TILE_SIZE)
+				glUseProgram(shader.Object());
+				glBindVertexArray(vao);
+
+				// Modify zoom for the first parallax layer.
+				if(isParallax)
+					zoom = sqrt(baseZoom * STAR_ZOOM * pow(pass, 0.2));
+
+				float length = backgroundVel.Length();
+				Point unit = length ? backgroundVel.Unit() : Point(1., 0.);
+				// Don't zoom the stars at the same rate as the field; otherwise, at the
+				// farthest out zoom they are too small to draw well.
+				unit /= pow(zoom, .75);
+
+				float baseZoom = static_cast<float>(2. * zoom);
```

Comment:
dunno, but the same variable name twice in a function? one time double one time float? that's just confusing to me